plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/ui/navigator/model/AppEngineProjectElement.java

```diff
@@ -109,12 +145,20 @@ static boolean hasLayoutChanged(Collection<IFile> changedFiles) {
    */
   private IFile descriptorFile;
 
-  private AppEngineDescriptor descriptor;
+  private String projectId;
+  private String projectVersion;
+  private String serviceId;
+
+  /** Return the App Engine environment type (e.g., standard, flexible). */
+  private String environmentType;
+
+  /** Return the App Engine runtime (e.g., java7, java8). */
```

Comment:
The comment is not really correct. The values are not compatible between standard and flex, and I think we should clearly explain it. For example, for standard, the values are `null` and `java8`, and for flex, the values are `java`, `custom`, `python27`, etc. I haven't checked the actual `runtime:` value of `app.yaml` auto-generated from `appengine-web.xml` where `<runtime>` is not present (i.e., java7) or it is set to `java8`, but another point of confusion could be the possibly different values of `runtime:` and `<runtime>` after the auto-conversion.

Reply:
Verified auto-generated `app.yaml` from an `appengine-web.xml` has `runtime: java7` or `runtime: java8`.